syntax = "proto3";

package api;
option go_package = "a.yandex-team.ru/alice/gamma/sdk/api";


message CardHeader {
    string text = 1;
}

message CardButton {
    string text = 1;
    bytes payload = 2;
    string url = 3;
}

message CardItem {
    string imageId = 1;
    string title = 2;
    string description = 3;
    CardButton button = 4;
}

message CardFooter {
    string text = 1;
    CardButton button = 2;
}

message Card {
    string type = 1;
    string imageId = 2;
    string title = 3;
    string description = 4;
    CardButton button = 5;
    CardHeader header = 6;
    repeated CardItem items = 7;
    CardFooter footer = 8;
}
